compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -29,6 +31,13 @@ object TreePickler {
       if isTermHole then s"{{{ $idx |" ~~ printer.toTextGlobal(tpe) ~~ "|" ~~ printer.toTextGlobal(args, ", ") ~~ "}}}"
       else s"[[[ $idx |" ~~ printer.toTextGlobal(tpe) ~~ "|" ~~ printer.toTextGlobal(args, ", ") ~~ "]]]"
   }
+
+  /** Select tree cannot be unambiguously resolved with signature alone,
+   *  needs to be pickled using the SELECTin tag. This is set by Typer, if
+   *  after overloading resolution it is discovered that other alternatives
+   *  have the same signature as the one that was selected.
+   */
+  val NeedsSelectIn = new Property.StickyKey[Unit]
```

Comment:
I'm fairly skeptical of this scheme, I think it can easily miss stuff, like when we explicitly select an overload without going through adaptation. This is not very common in the typer itself, but it could happen in a user-written macro. What's the performance impact of doing the check? What if we always used SELECTin instead of regular SELECT so we don't have to check anything?

Reply:
I thought of that, but that means we add one field per select node (which is the most common type of node of them all).

Reply:
Also, it does not work for structural refinements since there is no symbol to select on.